pkg/hydrate/controller_test.go

```diff
@@ -0,0 +1,170 @@
+				if err != nil {
+					t.Fatal(err)
+				}
+			}()
+
+			commitDir := filepath.Join(tempDir, tc.commit)
+			err = os.Mkdir(commitDir, os.ModePerm)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			kustFileGenerated, err := ioutil.TempFile(commitDir, "kustomization.yaml")
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = os.Rename(kustFileGenerated.Name(), filepath.Join(commitDir, "kustomization.yaml"))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// create a symlink to point to the temporary directory
+			dir := ft.NewTestDir(t)
+			symDir := dir.Root().Join(cmpath.RelativeSlash("run-hydrate-symlink"))
+			err = os.Symlink(commitDir, symDir.OSPath())
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer func() {
+				err := os.Remove(symDir.OSPath())
+				if err != nil {
+					t.Fatal(err)
+				}
+			}()
+
+			hydrator := &Hydrator{
+				DonePath:        "",
+				SourceType:      v1beta1.HelmSource,
+				SourceRoot:      cmpath.Absolute(commitDir),
+				HydratedRoot:    "",
+				SourceLink:      "",
+				HydratedLink:    "",
+				SyncDir:         "",
+				PollingPeriod:   1 * time.Minute,
+				RehydratePeriod: 1 * time.Minute,
+				ReconcilerName:  "root-reconciler",
+			}
+
+			absSourceDir := hydrator.SourceRoot.Join(cmpath.RelativeSlash(hydrator.SourceLink))
+			_, syncDir, err := SourceCommitAndDir(hydrator.SourceType, absSourceDir, hydrator.SyncDir, hydrator.ReconcilerName)
+			if err != nil {
+				t.Fatal(fmt.Errorf("failed to get commit and sync directory from the source directory %s: %v", commitDir, err))
+			}
+
+			err = hydrator.runHydrate(originCommit, syncDir.OSPath())
+			testutil.AssertEqual(t, tc.wantedErr, err)
+		})
+	}
+}
+
+func TestComputeCommit(t *testing.T) {
+	testCases := []struct {
+		name         string
+		sourceCommit string
+	}{
+		{
+			name:         "Computed commit should be the same to the one given in sourceDir",
+			sourceCommit: originCommit,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// create a temporary directory with a commit hash
+			tempDir, err := ioutil.TempDir(os.TempDir(), "compute-commit-test")
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer func(path string) {
+				err := os.RemoveAll(path)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}(tempDir)
+			absTempDir := cmpath.Absolute(tempDir)
+
+			commitDir := filepath.Join(tempDir, originCommit)
+			err = os.Mkdir(commitDir, os.ModePerm)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// create a symlink to point to the temporary directory
+			dir := ft.NewTestDir(t)
+			symDir := dir.Root().Join(cmpath.RelativeSlash("compute-commit-symlink"))
+			err = os.Symlink(commitDir, symDir.OSPath())
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer func() {
+				err := os.Remove(symDir.OSPath())
+				if err != nil {
+					t.Fatal(err)
+				}
+			}()
+
+			absSourceDir := absTempDir.Join(cmpath.RelativeSlash(tc.sourceCommit))
+			computed, err := ComputeCommit(absSourceDir)
```

Comment:
I think the input to this function should be `symDir`, right?

Reply:
Fixed!